extensions/wikia/PageHeader/ActionButton.class.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use CuratedContentHelper;
+
+class ActionButton {
+	const LOCK_ICON = 'wds-icons-lock-small';
+	const EDIT_ICON = 'wds-icons-pencil-small';
+
+	private $contentActions;
+	private $buttonAction;
+	private $pageStatsService;
+	private $title;
+	private $user;
+	private $request;
+	private $buttonIcon = self::EDIT_ICON;
+
+	public function __construct( \RequestContext $requestContext ) {
+		$skinVars = \F::app()->getSkinTemplateObj()->data;
+		$this->contentActions = $skinVars['content_actions'];
+
+		$this->title = $requestContext->getTitle();
+		$this->user = $requestContext->getUser();
+		$this->request = $requestContext->getRequest();
+
+		$this->pageStatsService = \PageStatsService::newFromTitle( $this->title );
+
+		$this->prepareActionButton();
+	}
+
+	public function getTitle(): \Title {
+		return $this->title;
+	}
+
+	public function getButtonAction(): array {
+		$this->buttonAction['data-tracking'] = $this->buttonAction['id'];
+		$this->buttonAction['icon'] = $this->buttonIcon;
+
+		return $this->buttonAction;
+	}
+
+	public function getDropdownActions(): array {
+		// items to be added to "edit" dropdown
+		$actions = [
+			'edit',
+			've-edit',
+			'formedit',
+			'history',
+			'move',
+			'protect',
+			'unprotect',
+			'delete',
+			'undelete',
+			'replace-file',
+			'talk',
+			'edit-mobile-main-page'
+		];
+
+		// Enable to modify actions list on dropdown
+		wfRunHooks( 'PageHeaderDropdownActions', [ &$actions ] );
+
+		return array_map( function( $action ) {
+			$ret = $this->contentActions[$action];
+			if ( isset( $ret['id'] ) ) {
+				$ret['data-tracking'] = $ret['id'] . '-dropdown';
+			}
+
+			return $ret;
+		}, array_filter( $actions, function( $action ) {
+			return isset( $this->contentActions[$action] );
+		}));
+	}
+
+	private function prepareActionButton() {
+		wfRunHooks( 'BeforePrepareActionButtons', [ $this, &$this->contentActions ] );
+
+		$isDiff = !is_null( $this->request->getVal( 'diff' ) );
+
+		// "Add topic" action
+		if ( isset( $this->contentActions['addsection'] ) ) {
+			// remove on diff pages (RT #72666)
+			if ( $isDiff ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)